### ISSUE TITLE
Best model output base on metric with evaluate_during_training

### ISSUE THREAD

**Issue by hawktang** [ThilinaRajapakse/simpletransformers#1281]
**Is your feature request related to a problem? Please describe.**
Best model output did not use evaluate_during_training metric to choose best output model

**Describe the solution you'd like**
Best model output base on metric with evaluate_during_training when evaluate_during_training enabled

**Describe alternatives you've considered**
I have read you source code and found below is working. But much complicated than expected. And use_early_stopping has to be eabled.

![image](https://user-images.githubusercontent.com/2004071/136430336-c299ee39-f7f4-46c2-810f-798eb1a8c6bb.png)

**Additional context**
Want a neat solution like

model_args = ClassificationArgs(
    num_train_epochs=10,
    train_batch_size=batch_size,
    eval_batch_size=batch_size,
    learning_rate=learning_rate,
    overwrite_output_dir=True,
    n_gpu=len(os.environ["CUDA_VISIBLE_DEVICES"].split(",")),
    evaluate_during_training=True,

---
    evaluate_during_training_metric='mcc',
---
    output_dir=output_dir,
    best_model_dir=best_model_dir,
    use_multiprocessing=True,
    max_seq_length=256,
    evaluate_during_training_verbose=True,
)

# Create a ClassificationModel
model = ClassificationModel(
    model_type,
    'roberta-base',
    num_labels=le.classes_.shape[0],
    weight=weight,
    args=model_args,
    use_cuda=True,
)

# Train the model
model.train_model(train_df=train_df, eval_df=eval_df, show_running_loss=True)
